python/ray/dashboard/node_stats.py

```diff
@@ -91,6 +92,55 @@ def get_node_stats(self):
                 key=itemgetter("boot_time"))
             return {"clients": node_stats}
 
+    # Gets actors in a flat way to allow for grouping by actor type.
+    def get_actors(self, workers_info_by_node, infeasible_tasks, ready_tasks):
+        now = time.time()
+        actors: Dict[str, Dict[str, any]] = {}
+        # construct flattened actor tree
+        with self._node_stats_lock:
+            for addr, actor_id in self._addr_to_actor_id.items():
+                actors[actor_id] = copy.deepcopy(self._default_info)
+                actors[actor_id].update(self._addr_to_extra_info_dict[addr])
+
+            for node_id, workers_info in workers_info_by_node.items():
+                for worker_info in workers_info:
+                    if "coreWorkerStats" in worker_info:
+                        core_worker_stats = worker_info["coreWorkerStats"]
+                        addr = (core_worker_stats["ipAddress"],
+                                str(core_worker_stats["port"]))
+                        if addr in self._addr_to_actor_id:
+                            actor_info = actors[self._addr_to_actor_id[addr]]
+                            format_reply_id(core_worker_stats)
+                            actor_info.update(core_worker_stats)
+                            actor_info["averageTaskExecutionSpeed"] = round(
```

Comment:
Isn't this ignoring the fact that there is time when actors are idle?

Reply:
Yes, it is, you're right. However, this is the same behavior we currently have on the dashboard. Maybe we should brainstorm a way to capture only the time when the actor is not idle.

Reply:
Yeah that sounds good to me!